### PR TITLE
Update assited-service lint and verify-generated-code to centos9

### PR DESCRIPTION
**Pull request by michaellevy101**

### CODE REVIEW

**Review by nmagnezi — COMMENTED** [review]
Looks okay to me, but @osherdp should probably review as well.

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml [hunk 1]
```diff
@@ -818,7 +846,6 @@ tests:
     cluster_profile: packet-assisted
     dependencies:
       HYPERSHIFT_IMAGE: hypershift-operator
-      PROVIDER_IMAGE: cluster-api-provider-agent
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:821]: why did you remove it?
it's especially important to have it before next FF to make sure we're not mistakenly using the wrong branch of capi-provider-agent

File: ci-operator/step-registry/assisted/verify-generated-code-v2/assisted-verify-generated-code-v2-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+ref:
+  as: assisted-verify-generated-code-v2
+  from: assisted-service-generate
```
Comment (osherdp) [inline ci-operator/step-registry/assisted/verify-generated-code-v2/assisted-verify-generated-code-v2-ref.yaml:3]: the only difference between ``assisted-verify-generated-code`` and ``assisted-verify-generated-code-v2`` is this line of code?
because if so we don't need to duplicate this step. As long as the image does the same thing we can do (in ``release-ocm-2.5`` configuration file):
```
images:
...
- dockerfile_literal: |
    FROM base
  from: assisted-service-build
  to: assisted-service-generate
```
and the step will run with no problems

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml [hunk 3]
```diff
@@ -111,6 +115,30 @@ images:
...
+        /assisted-service/hack/setup_env.sh jq && \
+        /assisted-service/hack/setup_env.sh kustomize && \
+        go install github.com/golang/mock/mockgen@v1.5.0 && \
+        go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.6.2 && \
+        go install golang.org/x/tools/cmd/goimports@v0.1.5 && \
+        chmod 775 -R $GOPATH
+    RUN curl --retry 5 -L https://github.com/operator-framework/operator-sdk/releases/download/v1.10.1/operator-sdk_linux_amd64 --output /tmp/operator-sdk_linux_amd64 && \
+        chmod +x /tmp/operator-sdk_linux_amd64 && \
+        install /tmp/operator-sdk_linux_amd64 /usr/local/bin/operator-sdk
+    RUN python3 -m venv ${VIRTUAL_ENV:-/opt/venv} && \
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:133]: it seems redundant to have a default value when we clearly defined it above (with ``ENV VIRTUAL_ENV=/opt/venv``), no?

**Review by osherdp — COMMENTED** [review]
Please next time have it holded and/or add some info in the description of the PR. it's really hard to get a grasp of the changes when you don't include a proper description...
